src/services/namex.services.ts

```diff
@@ -273,6 +273,35 @@ export default class NamexServices {
     }
   }
 
+  static async getNameRequestByToken (handleError: boolean, nrId:string, accountId:string): Promise<NameRequestI> {
+    try {
+      const token = sessionStorage.getItem('KEYCLOAK_TOKEN')
+      if (token) {
+        const headers = {
+          Authorization: `Bearer ${token}`,
+          'Accept': 'application/pdf'
+        }
+
+        // call namerequest get endpoint to retrieve NR detail, also verify affiliation
+        const response = await this.axios.get(`${this.namexUrl()}/namerequests/${nrId}?org_id=${accountId}`, {
+          headers
+        })
+        if (response?.status === OK && response?.data) return response.data
+        throw new Error(`Invalid response = ${response}`)
+      } else {
+        // if without token then check the session storage
+        return await this.getNameRequest(handleError)
```

Comment:
I don't understand how this "checks the session storage". It looks like it does the same GET as above but without the org_id (and it does something with a cancel token).

If you figure out exactly how this works, please add comments.

Reply:
In business dashboard (auth-web), it will set the NR number, email etc information into session storage and the namerequest will search NR details by for these information. However, when we migrate to new domain, this way doesn't work. The token (cancel token) is never been used in the namerequests get endpoint.

![image](https://user-images.githubusercontent.com/44277752/234989302-a088a273-8382-42e7-bbe6-5fc372cbabf7.png)

Reply:
not right now. The namex api need to be upgrade completely.

Reply:
OK